gum/backend-x86/guminterceptor-x86.c

```diff
@@ -216,9 +225,19 @@ _gum_interceptor_backend_activate_trampoline (GumInterceptorBackend * self,
 
   gum_x86_writer_reset (cw, prologue);
   cw->pc = GPOINTER_TO_SIZE (ctx->function_address);
-  gum_x86_writer_put_jmp_address (cw, GUM_ADDRESS (ctx->on_enter_trampoline));
+
+  if (ctx->type == GUM_INTERCEPTOR_TYPE_FAST)
+  {
+    gum_x86_writer_put_jmp_address (cw, GUM_ADDRESS (ctx->replacement_function));
+  }
+  else
+  {
+    gum_x86_writer_put_jmp_address (cw, GUM_ADDRESS (ctx->on_enter_trampoline));
+  }
```

Comment:
```suggestion
    gum_x86_writer_put_jmp_address (cw, GUM_ADDRESS (ctx->replacement_function));
  else
    gum_x86_writer_put_jmp_address (cw, GUM_ADDRESS (ctx->on_enter_trampoline));
```

Another issue is that the first statement exceeds 80 cols and needs further tweaks. (Too risky/tedious to suggest this change from the GitHub UI.)